snuba/utils/streams/dummy.py

```diff
@@ -75,17 +75,28 @@ def poll(self, timeout: Optional[float] = None) -> Optional[Message[TPayload]]:
 
         # TODO: Throw ``EndOfPartition`` errors.
         for partition, offset in sorted(self.__offsets.items()):
+            messages = self.__messages[partition]
             try:
-                payload = self.__messages[partition][offset]
+                payload = messages[offset]
             except IndexError:
-                pass
+                if not offset == len(messages):
+                    raise ConsumerError("invalid offset")
```

Comment:
This basically emulates `auto.offset.reset` being set to `error` in the Kafka consumer. I still need to find a way to make this configuration (and controlling partition EOF behavior) generic enough that it can be passed to any consumer regardless of the specific implementation.